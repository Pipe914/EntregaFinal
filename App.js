import React, {Component} from "react";
import "react-native-gesture-handler";
import { NavigationContainer } from "@react-navigation/native";
import { createStackNavigator } from "@react-navigation/stack";

import HomeScreen from "./components/HomeScreen.js";
import WebViewComponent from "./components/WebViewComponent.js";
import { MovieProvider } from "./contexts/MovieContexts.js";
import MovieList from "./components/MovieList.js";
import AddMovieForm from "./components/MovieForm.js";
import MovieDetails from "./components/MovieDetails.js";

const Stack = createStackNavigator();

class App extends Component {

  render() {
    return (
      <MovieProvider>
        <NavigationContainer>
          <Stack.Navigator>
            <Stack.Screen name="Home" component={HomeScreen} />
            <Stack.Screen name="MovieList" component={MovieList} />
            <Stack.Screen name="AddMovie" component={AddMovieForm} />
            <Stack.Screen name="WebView" component={WebViewComponent} />
            <Stack.Screen name="MovieDetails" component={MovieDetails} />
          </Stack.Navigator>
        </NavigationContainer>
      </MovieProvider>
    );
  }
}

export default App;
