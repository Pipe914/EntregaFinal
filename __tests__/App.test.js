import React from 'react';
import {cleanup, render} from '@testing-library/react-native';
 
import App from '../App';
 
afterEach(cleanup);
 


describe('App_class', () => {

  // Tests that the App component renders with all screens.
  it("test_renders_app_with_all_screens", () => {
      const wrapper = shallow(<App />);
      expect(wrapper.find(MovieProvider)).toHaveLength(1);
      expect(wrapper.find(NavigationContainer)).toHaveLength(1);
      expect(wrapper.find(Stack.Navigator)).toHaveLength(1);
      expect(wrapper.find(Stack.Screen)).toHaveLength(5);
  });

  // Tests that the user can navigate to the MovieList screen.
  it("test_navigates_to_movie_list_screen", () => {
      const navigation = { navigate: jest.fn() };
      const wrapper = shallow(<HomeScreen navigation={navigation} />);
      wrapper.find(Pressable).at(0).simulate('press');
      expect(navigation.navigate).toHaveBeenCalledWith("MovieList");
  });

  // Tests that the MovieList screen handles an empty movie list.
  it("test_handles_empty_movie_list", () => {
      const navigation = { navigate: jest.fn() };
      const wrapper = shallow(<MovieList navigation={navigation} />);
      expect(wrapper.find(Text)).toHaveLength(1);
      expect(wrapper.find(Text).text()).toEqual("No tienes favoritos");
  });

  // Tests that the AddMovieForm screen handles invalid input.
  it("test_handles_invalid_input_in_add_movie_form", () => {
      const addFavoriteMock = jest.fn();
      const wrapper = shallow(<AddMovieForm addFavorite={addFavoriteMock} />);
      wrapper.find(Button).simulate('press');
      expect(addFavoriteMock).not.toHaveBeenCalled();
  });

  // Tests that the user can add a movie to the list.
  it("test_adds_movie_to_list", () => {
      const addFavoriteMock = jest.fn();
      const wrapper = shallow(<AddMovieForm addFavorite={addFavoriteMock} />);
      wrapper.find(TextInput).at(0).simulate('changeText', 'Test Movie');
      wrapper.find(TextInput).at(1).simulate('changeText', 'Test Director');
      wrapper.find(TextInput).at(2).simulate('changeText', '2022');
      wrapper.find(TextInput).at(3).simulate('changeText', 'Test Productora');
      wrapper.find(TextInput).at(4).simulate('changeText', 'Test Genero');
      wrapper.find(TextInput).at(5).simulate('changeText', '1000000');
      wrapper.find(Button).simulate('press');
      expect(addFavoriteMock).toHaveBeenCalled();
  });

  // Tests that the MovieDetails screen fetches movie details from Firebase.
  it("test_fetches_movie_details_from_firebase", () => {
      const route = { params: { name: "Test Movie" } };
      const wrapper = shallow(<MovieDetails route={route} />);
      expect(wrapper.find(Text)).toHaveLength(1);
      expect(wrapper.find(Text).text()).toEqual("Cargando...");
  });
});
