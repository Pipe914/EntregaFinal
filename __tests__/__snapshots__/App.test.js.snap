// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should show hello world 1`] = `
<View
  style={
    {
      "backgroundColor": "#ffffff",
    }
  }
>
  <View
    style={
      {
        "marginTop": 32,
        "paddingHorizontal": 24,
      }
    }
  >
    <Text
      style={
        {
          "color": "#000000",
          "fontSize": 24,
          "fontWeight": "600",
        }
      }
    >
      Hello World!
    </Text>
  </View>
</View>
`;
